pkg/containers/utils.go

```diff
@@ -0,0 +1,83 @@
+package containers
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"syscall"
+)
+
+// runsOnContainerV1 tests if process is running on a container in cgroupsv1
+// it tests by checking for existance of a release_agent file in cpuset
+// from https://www.kernel.org/doc/Documentation/cgroup-v1/cgroups.txt:
+//
+// - release_agent: the path to use for release notifications (this file exists in the top cgroup only)
+//
+// WARNING: IF THIS WASN'T CLEAR THIS IS ONLY VALID FOR CGROUPSV1 MACHINES
+func runsOnContainerV1() (bool, error) {
+	const releseAgentPath = "/sys/fs/cgroup/cpuset/release_agent"
+	_, err := os.Stat(releseAgentPath)
+	if os.IsNotExist(err) {
+		return true, nil
+	}
+	if err != nil {
+		return false, err
+	}
+	return false, nil
+}
+
+// mountsCpuset mounts /sys/fs/cgroup/cpuset to the container's cpuset fs
+// this is done in order to detect host cgroup data when running tracee from a container in cgroupsv1
+func mountCpuset(mountPath string) error {
+	isMp, _ := isMountpoint(mountPath)
+
+	if isMp {
+		return nil
+	}
+
+	// equivalent to mount -t cgroup -o cpuset cgroup /sys/fs/cgroup/cpuset
+	if err := syscall.Mount("cgroup", mountPath, "cgroup", 0, "cpuset"); err != nil {
+		return fmt.Errorf("failed to mount %s", err)
+	}
+	return nil
+}
+
+func isMountpoint(path string) (bool, error) {
+	mountpointCmd := fmt.Sprintf("mountpoint %s | grep \"is a mountpoint\"", path)
+	out, err := exec.Command("sh", "-c", mountpointCmd).Output()
+	if err != nil {
+		return false, err
+	}
```

Comment:
Let's avoid invoking external commands.
Instead, you can look for the path in `/proc/mounts`

Reply:
Invoking external commands is not a good security practice, although you are right that we control the image files.
Yet it is more perfomant to simply read the /proc/mounts file, and we already do that in procMountsCgroups() so you can simply use the same code, without invoking external commands.